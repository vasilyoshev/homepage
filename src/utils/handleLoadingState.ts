import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
