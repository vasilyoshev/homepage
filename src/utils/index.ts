export * from './handleLoadingState';
export * from './testUtils';
