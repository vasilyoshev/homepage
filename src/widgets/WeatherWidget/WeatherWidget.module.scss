@import url('https://fonts.googleapis.com/css2?family=Montserrat&display=swap');

$headers-font: 'Montserrat', sans-serif;

.weatherWidgetContainer {
  padding: 8px;
}

.title,
.locations {
  margin: 0 auto;
  font-family: $headers-font;
  text-align: center;
};
