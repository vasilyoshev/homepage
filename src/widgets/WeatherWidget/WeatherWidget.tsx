import React, {useState} from 'react';
import { WeatherWidgetLocationSearch, WeatherWidgetLocationTable } from 'components';

export const WeatherWidget: React.FC = () => {
  const [locations, setLocations] = useState<string[]>([]);
  const addLocation = (location: string) => setLocations([location, ...locations]);

  return (
    <main>
      <h1>Weather Widget</h1>
      <WeatherWidgetLocationSearch onSearch={addLocation} />
      <WeatherWidgetLocationTable locations={locations} />
    </main>
  );
};
