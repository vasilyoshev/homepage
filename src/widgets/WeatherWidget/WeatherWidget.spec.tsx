import { render } from 'utils';
import { WeatherWidget } from './WeatherWidget';

describe('WeatherWidget', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<WeatherWidget />);

    expect(asFragment()).toMatchSnapshot();
  });
});
