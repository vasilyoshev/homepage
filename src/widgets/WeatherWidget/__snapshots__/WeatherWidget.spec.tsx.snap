// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherWidget renders successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h1
        class="title"
      >
        Weather Widget
      </h1>
      <div
        class="MuiPaper-root MuiCard-root searchContainer MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                placeholder="add location here"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-2"
                  style="width: 0.01px;"
                >
                  <span>
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Search
              </span>
            </button>
          </div>
        </div>
      </div>
      <h2
        class="title"
      >
        Locations
      </h2>
      <div
        class="MuiTableContainer-root tableContainer"
      >
        <table
          aria-label="a dense table"
          class="MuiTable-root"
        >
          <tbody
            class="MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
